docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,306 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Super Simple Discovery API
+
+## Overview
+
+It is surprisingly hard to figure out how to talk to a Kubernetes cluster.  Not only do clients need to know where to look on the network, they also need to identify the set of root certificates to trust when talking to that endpoint.
+
+This presents a set of problems:
+* It should be super easy for users to configure client systems with a minimum of effort `kubectl` or `kubeadm init` (or other client systems).
+  * Establishing this should be doable even in the face of nodes and master components booting out of order.
+  * We should have mechanisms that don't require users to ever have to manually manage certificate files.
+* Over the life of the cluster this information could change and client systems should be able to adapt.
```

Comment:
1. perhaps, but do admis write those files by hand or what do they use? do they really have to run a service behind that URL?
2. that's a thing, and people still do such things over email, and some people just use git and pretend it's secure
3. JWS server is currently not integrated into `apiserver`, so we could just keep it that way, if there are great many reasons to keep it optional